src/clean.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as chalk from 'chalk';
+import * as pify from 'pify';
+import * as rimraf from 'rimraf';
+
+import {Options} from './cli';
+import * as tscfg from './tscfg';
+
+const rimrafp = pify(rimraf);
+
+/**
+ * Remove files generated by the build.
+ */
+export async function clean(options: Options): Promise<void> {
+  const tsconfig = await tscfg.getTsConfig(options.targetRootDir);
+  if (tsconfig.compilerOptions && tsconfig.compilerOptions.outDir) {
```

Comment:
We might want to guard against `outDir` being `.` as well. That's going to be quite disastrous. I guess we require that `outDir` should not have anything that you would like to keep. `.` is going to be a common mistake / misunderstanding.